linera-service/src/server.rs

```diff
@@ -148,25 +151,32 @@ impl ServerContext {
                 }
             });
         }
+
         join_all(handles).await;
 
         Ok(())
     }
 
-    fn start_metrics(host: &str, port: u16) {
+    fn start_metrics(host: &str, port: &u16) {
         match format!("{}:{}", host, port).parse::<SocketAddr>() {
-            Err(err) => error!("Invalid metrics address: {err}"),
-            Ok(address) => {
-                if let Err(error) = metrics_exporter_tcp::TcpBuilder::new()
-                    .listen_address(address)
-                    .install()
-                {
-                    error!(
-                        ?error, %address,
-                        "Could not install TCP metrics exporter."
-                    );
-                }
-            }
+            Err(err) => error!("Invalid metrics address for {host}:{port}: {err}"),
+            #[cfg(not(feature = "prometheus-metrics"))]
+            Ok(address) => Self::start_metrics_impl(address),
+            #[cfg(feature = "prometheus-metrics")]
+            Ok(address) => prometheus_server::start_metrics(address),
```

Comment:
Do we need to save the returned future here, and call join on it with the actual server's future also? 🤔

Reply:
What future?

Reply:
Do you mean the handle?

Reply:
Yes, the handle returned by spawn. Do we need to join it with the server one?

Reply:
Probably not if it never terminates

Reply:
Oh ok, I asked because the server one doesn't terminate either but we still wait on it. But I guess it doesn't matter anyways. Thanks!